distributed/asyncio.py

```diff
@@ -146,11 +135,9 @@ def __del__(self):
         if self.status == 'running':
             return
 
-        if self._make_current:
-            self.loop.make_current()
+        self.status = 'running'
```

Comment:
Set status before starting to connect. Fixes race if several coroutines await client.start() with the same client concurrently.